### DISCUSSION TITLE
Boots to apple monitor, not “waiting for RPI..”

### DISCUSSION THREAD

**Discussion by microbmen** [tjboldt/Apple2-IO-RPi#139] [General]
Wondering if I have a chip issue or a firmware issue. 

When I boot the card drops into apple monitor.  Sometimes I see Apple2-IO-RPI before it drops to monitor.  

![image](https://user-images.githubusercontent.com/2065443/227808631-8e00a70c-4bb3-47c9-885f-4d2698b2d2fc.jpeg)

**Comment by tjboldt** [comment]
Some things that have caused issues in the past:

1. Using a 74LVC245 where the 74LS245 goes
2.  Missing capacitors
3. Cold solder joint
4. Bad 28C64B

**Reply by microbmen** [reply]
I have 2 cards.  One bought fully assembled on eBay. The other was a blank board I printed and populated myself.  
Both boards doing the same behavior.   I tried reflashing the firmware from a fresh download.  Same thing. 
I can get as far as 
![image](https://user-images.githubusercontent.com/2065443/228016555-1d3639b6-fec0-4375-88a1-7e11761adfdd.jpeg)

Sometimes.  

I think there must be a conflict with something in my machine.  I'll try again when I have some time.

**Reply by tjboldt** [reply]
That also happened with another known working board in a system with other cards that draw a lot of current. There is only 500 mA available to all cards combined and the RPi Z 2 W draws half of that alone.

**Reply by microbmen** [reply]
I thought of that and powered the Rpi from outside power.  Same issue.   I need more time to experiment but this system has a RomX onboard.  A few times on booting some visual artifacts from the RomX boot menu came up when it crashed.  I thought that was suspicious.

**Reply by microbmen** [reply]
Tried it on another apple iie and worked perfect.  
I have to isolate what card is causing this and  let you know.   I suspect it's the RomX thou.  
Slot 7 - RPi
Slot 6 - Yellowstone
Slot 5 -
Slot 4 - Phasor
Slot 3 -Uthernet ii
Slot 2 - Super Serial 
Slot 1 - Grappler +
Slot 0 - RamWorks

RomX in all 3 ROMs

**Reply by microbmen** [reply]
would you be able to look through here to see if there are any memory conflicts?  I think the ROMX is the issue.  Not sure if anyone tested with that yet? 

https://theromexchange.com/documentation/romxce/ROMXce%20API%20Reference.pdf

specifically


Other Hardware Addresses
Since	the	ROMXc	and	ROMXe	banks	can	hold	32K	but	only	map	to	the	roughly	16K	ROM	space	at	$C1xx-$FFFF,	
each	bank	is	sub-divided	into	a	lower	and	upper	bank.	This	is	similar	to	how	the	Apple	//c	System	ROMs	(except	
the	ROM	255)	are	mapped.	When	in	the	Main	operating	mode,	selection	of	the	sub-banks	is	handled	in	the	usual	
fashion	by	the	RA14	signal	from	the	IOU	chip.	In	the	ROMXe	– and	while	in	the	Bank	0	firmware	– sub-bank	
selection	is	handled	by	the	following	softswitches:
$F820 (Selects Main ROM Lower Bank)
$F821 (Selects Main ROM Upper Bank)
$F822 (Selects Text ROM Lower Bank; currently not implemented)
$F823 (Selects Text ROM Upper Bank; currently not implemented)
$F824 (Selects Temporary ROM Lower Bank)
$F825 (Selects Temporary ROM Upper Bank)
Reserved	addresses
The	following	addresses	are	Reserved	for	ROMX	firmware	use	and	SHOULD	NEVER	BE	ACCESSED	while	Bank	0	is	
active:
$C7xx (Reserved)
$F826-F (Reserved)
$F87x (Reserved)

**Reply by tjboldt** [reply]
If the Apple2-IO-RPi card is in slot 7, it will use the $C7xx space which will conflict with ROMX firmware.

**Reply by microbmen** [reply]
@tjboldt is there something needed to get the card to work in slots other than slot 7?  In a machine without RomX, the card works flawless in slot 7.  However, when I put the card in any other slot,  I tired slot 6, slot 5, slot 4.  It never loads.  Tried the corresponding PR#6, PR#5, PR#4, etc. to boot from the alternate slots and nothing happens.  The card only wants to work in slot 7.

**Reply by tjboldt** [reply]
Did you program the EEPROM with an external programmer or via the Apple II itself? If you did it in-system from slot 7, it only has the firmware for slot 7. You'll need to run the utility again for the slot you want to use (but only need to do this once per slot as it retains the firmware space for the other slots after writing).

**Reply by microbmen** [reply]
Interesting.  ok, yes I programmed via Apple II itself.

**Reply by microbmen** [reply]
Got it to boot in other slots, i.e. slot 5 by writing the eeprom while in slot 5.   however, still conflicts with RomX in other slots.

**Reply by tjboldt** [reply]
That is strange because the card firmware should be active whenever I/O SELECT line goes low on that particular slot. If ROMX is putting data on the bus when the I/O SELECT line is low for a slot, that is bug in the ROMX. Of course, I'd expect it to fail for all cards, not just this one so maybe there is something else wrong here. From what I read in the Apple ][ Reference Manual, the enable lines I've used should be correct. For example, in a IIgs, the card firmware is only active when "Your Card" is selected for the slot. If the IIgs firmware is used, the I/O SELECT line stays high and the card does not interfere.
